### PR TITLE
chore: clean up forward declarations

### PR DESCRIPTION
**Pull request by dsanders11**
#### Description of Change

Cleans up forward declarations:
* Trims some unused ones
* Adds some new ones
* ~Puts the trailing `// namespace ...` comment on forward declaration namespaces which were missing it~

The removing unused and adding new ones were guided by `iwyu`'s output, but wasn't able to take it at face value as sometimes it made confusing suggestions.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes

#### Release Notes

Notes: none

### CODE REVIEW

File: chromium_src/chrome/browser/process_singleton.h [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
 namespace base {
 class CommandLine;
-}
+}  // namespace base
```
Comment (zcbenz) [inline chromium_src/chrome/browser/process_singleton.h:31]: Majority of forward declarations in Chromium seem to not include the `// namespace xxx`, but I could not find any documented rules about this, and there are some places having mixed style:
https://source.chromium.org/chromium/chromium/src/+/main:content/public/browser/web_contents.h;l=51-80

But I'm good unifying the style.

**Review by zcbenz — APPROVED** [review]

**Review by nornagon — COMMENTED** [review]
personally I prefer not to have the trailing `// namespace`, especially when the contents of the brace pair is so short.

**Comment by dsanders11** [comment]
I don't really have an opinion either way on the `// namespace`, just looking to make it consistent in the code base. Since @zcbenz found that most forward declarations in Chromium don't use it, and @nornagon isn't a fan, I'll invert this and remove the ones which have it, so that it's consistent.

**Comment by dsanders11** [comment]
It looks like the source of the inconsistencies with the namespace end comments is `clang-format`, it started fighting me when I tried to remove some of them. The check `FixNamespaceComments` adds them to namespace blocks which aren't short, which is currently hardcoded in `clang-format` to be defined as 1 line, so it adds the comments to any forward declarations with more than one line. That can lead to kind of funky results, like a forward declaration in nested namespaces will get an end comment added for the outer namespace but not the inner one. Unfortunate that that check has some quirks.

Looks like `ShortNamespaceLines` was just landed in `clang-format` a few months ago to remove the hardcoding and allow it to be customized. We may want to revisit once that makes its way into the version of `clang-format` being used in Electron, and bump it to a less annoying value.

**Pull request merged**